### DISCUSSION TITLE
Welcome to PinLayout Discussions!

### DISCUSSION THREAD

**Discussion by lucdion** [layoutBox/PinLayout#222] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by KatyaVol** [comment]
Hello, my name is Katya, and i am iOS developer.

Our team is currently integrating your library, and we are unsure about the comment regarding Automatic Sizing. Is Automatic Sizing still in beta for iOS, or has it been officially released? We have a policy that restricts the use of beta versions in production code, so we would like to clarify whether this note is still relevant.

Your notes for Automatic Sizing:
1. Automatic Sizing is currently only available on iOS.
2. Automatic Sizing is still in beta, so any comments are welcomed.

Thank you!

**Reply by lucdion** [reply]
It's now officially released, but to be 100% transparent, it's one of the least used features and, therefore, the least tested. I'm sorry I can't be more specific, but I prefer to be open.

**Reply by KatyaVol** [reply]
Thank you so much for a quick reply)))

**Comment by KatyaVol** [comment]
@lucdion 
Hi, I see two similar examples: in one case, the setNeedsLayout function is called, and in the other, it is not. My question is—why is that? In what case should setNeedsLayout() be called in the configure() function, and in what case is it not needed?

Should setNeedsLayout() be called in configure() if the parameter changes require recalculating the layout? If only the data changes without affecting the size, is setNeedsLayout() unnecessary?

Thank you!

![NO_setNeedsLayout](https://github.com/user-attachments/assets/4c16715b-9674-48f2-b509-a8b9c9a4b583)
![setNeedsLayout_called](https://github.com/user-attachments/assets/cf31d533-19e1-4805-8901-c14a163dddd4)
